File: BOJ/구현/10814(나이순 정렬)/10814(나이순 정렬).java
import java.io.*;
import java.util.ArrayList;
import java.util.Collections;

public class Main {

    static class Member implements Comparable<Member>{
        int age;
        String name;
        int idx;

        Member(int age, String name, int idx) {
            this.age = age;
            this.name = name;
            this.idx = idx;
        }

        @Override
        public int compareTo(Member o) {
            if(this.age < o.age) return -1;
            else if(this.age > o.age)    return 1;
            else{
                if(this.idx < o.idx) return -1;
                else return 1;
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        ArrayList<Member> list = new ArrayList<Member>();
        int N = Integer.parseInt(br.readLine());
        for(int i=0;i<N;i++){
            String[] buf = br.readLine().split(" ");
            Member m = new Member(Integer.parseInt(buf[0]), buf[1], i);
            list.add(m);
        }
        Collections.sort(list);

        for(Member m : list){
            bw.write(m.age + " " + m.name + "\n");
        }
        bw.flush();
    }
}

File: BOJ/구현/10814(나이순 정렬).cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

struct Member {
	int age;
	string name;
	int idx;
};

int main() {
	ios_base::sync_with_stdio(false);	cin.tie(NULL);
	vector<Member> list;
	int N;
	cin >> N;
	
	for(int i=0;i<N;i++) {
		int age;	string name;
		cin >> age >> name;
		list.push_back({ age, name, i });
	}

	sort(list.begin(), list.end(), [](Member &a, Member &b) {
		if (a.age != b.age)	return a.age < b.age;
		else return a.idx < b.idx;
	});

	for (auto it = list.begin(); it != list.end(); it++) {
		cout << it->age << " " << it->name << "\n";
	}
	return 0;
}